### PR TITLE
STAC serializer (renderer) for RasterMetaEntry

### PR DESCRIPTION
**Pull request by mcovalt**
It describes the raster's 

- footprint
- bounding box
- related images
- ancillary files
- projection

E.g.:
```json
{
   "type":"Feature",
   "stac_version":"1.0.0-beta.2",
   "id":139,
   "properties":{
      "proj:epsg":32618,
      "proj:transform":[
         499980.0,
         20.0,
         0.0,
         4800000.0,
         0.0,
         -20.0
      ],
      "datetime":"2021-05-06T18:05:45.018503Z"
   },
   "geometry":{
      "type":"Polygon",
      "coordinates":[
         [
            [
               -75.00024679052602,
               43.35285539188815
            ],
            [
               -73.6455526700249,
               43.344834816006944
            ],
            [
               -73.66698539120085,
               42.356322389620374
            ],
            [
               -75.00024288390158,
               42.36407191996271
            ],
            [
               -75.00024679052602,
               43.35285539188815
            ]
         ]
      ]
   },
   "links":[
      
   ],
   "assets":{
      "image-133":{
         "href":"http://localhost:9000/rgd-storage/1f9254bc-ee20-4914-a944-8d04f5aee120/L1C_T18TWN_A028079_20201106T160014.tif?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=mcovalt%2F20210506%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210506T195304Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=c75bb4d1194d69f5ffdad6c4c8185ef4d4cc3e896cc49aa566ef6fa0648e19fe",
         "title":"L1C_T18TWN_A028079_20201106T160014.tif"
      }
   },
   "bbox":[
      499980.0,
      4690200.0,
      609780.0,
      4800000.0
   ],
   "stac_extensions":[
      "projection"
   ]
}
```

### CODE REVIEW

**Comment by aashish24** [comment]
we will need link to downloadable asset, relationship to category or collection probably not at this stage. 

@matt-bernstein could you test this / provide feedback?

**Comment by banesullivan** [comment]
> Do we want to link to a downloadable assets?

Yes, I think we do. For rasters, we want to link all related `ImageFile`s and all `ChecksumFile`s in the `ancillary_files` field.

File: rgd/geodata/serializers.py [hunk 1]
```diff
@@ -212,4 +213,16 @@ class Meta:
         fields = '__all__'
 
 
+class STACRasterSerializer(serializers.BaseSerializer):
+    def to_representation(self, instance: models.RasterMetaEntry) -> dict:
+        item = pystac.Item(
+            id=instance.pk,
+            geometry=json.loads(instance.footprint.json),
+            bbox=instance.footprint.extent,
```
[the remark this answers is not in the document]
Comment (banesullivan) [reply rgd/geodata/serializers.py:221]: Actually, I get what this is doing. Probably best to use `footprint`

[the remark this answers is not in the document]
Comment (mcovalt) [reply rgd/geodata/serializers.py:221]: I think footprint is what we're looking for. See [here](https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#item-fields):

> Defines the full footprint of the asset represented by this item.

An 'item' is essentially the footprint of all the underlying 'assets'.

File: rgdc/rgdc/rgdc.py [hunk 2]
```diff
@@ -121,6 +121,14 @@ def download_raster_thumbnail(
...
             raise IndexError(f'Band index ({band}) out of range.')
 
+    def get_raster_stac(self, raster_meta_id: Union[str, int, dict]) -> Dict:
+        """Get raster entry as STAC item dictionary/JSON."""
+        if isinstance(raster_meta_id, dict):
+            raster_meta_id = spatial_subentry_id(raster_meta_id)
+
+        r = self.session.get(f'geodata/imagery/raster/{raster_meta_id}/stac')
+        return r.json()
+
```
Comment (banesullivan) [inline rgdc/rgdc/rgdc.py:131]: @AlmightyYakob, would you please review?

**Comment by banesullivan** [comment]
I say once we get cloud coverage in this PR, let's merge it. Then I will follow up with a separate PR to add a new search endpoint that will resolve both #374 and #352

File: rgdc/rgdc/rgdc.py (hunk 2 above)
Comment (banesullivan) [reply rgdc/rgdc/rgdc.py:131]: I like this idea, will implement

**Review by AlmightyYakob — APPROVED** [review]

**Pull request merged**